packages/test/functional-tests/src/test/ddsHandleEncoding.spec.ts

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	MockFluidDataStoreRuntime,
+	MockStorage,
+	MockDeltaConnection,
+	MockHandle,
+} from "@fluidframework/test-runtime-utils";
+import { DirectoryFactory, IDirectory, MapFactory } from "@fluidframework/map";
+import { IChannel, IChannelFactory } from "@fluidframework/datastore-definitions";
+import { ConsensusRegisterCollectionFactory } from "@fluidframework/register-collection";
+import { detectOutboundReferences } from "@fluidframework/container-runtime";
+
+/**
+ * The purpose of these tests is to demonstrate that DDSes do not do opaque encoding of handles
+ * when preparing the op payload (e.g. prematurely serializing).
+ * This is important because the runtime needs to inspect the full op payload for handles.
+ */
+describe("DDS Handle Encoding", () => {
+	const handle = new MockHandle("whatever");
+	const messages: any[] = [];
+
+	beforeEach(() => {
+		messages.length = 0;
+	});
+
+	/**
+	 * @returns The list of handles found in the given contents object
```

Comment:
nit: add more details here. Something that says it uses the logic used by container runtime to detect handles for GC?